tree.py

```diff
@@ -62,12 +63,16 @@
 from reynir.binparser import BIN_Token
 from reynir.simpletree import SimpleTree, SimpleTreeBuilder, NonterminalMap, IdMap
 from reynir.cache import LRU_Cache
-from typing_extensions import TypedDict
+
+# Processing environment
+# A mapping of keywords or nonterminal names to functions
+# for processing sentence trees or, more specifically, query trees
+ProcEnv = MutableMapping[str, Any]
 
 
 class TreeStateDict(TypedDict, total=False):
-    session: Session
-    processor: ModuleType
+    session: Required[Session]
```

Comment:
Does this definitely work from Python 3.7 onwards? You don't need to import from typing_extensions?

Reply:
I import `Required` and `TypedDict` from `typing_extensions` :) I work on PyPy 7.3.9 which corresponds to Python 3.7.13, so this should be all good.